business_register/views/pep_views.py

```diff
@@ -22,3 +27,9 @@ def get_serializer_class(self):
         if self.action == 'retrieve':
             return PepDetailSerializer
         return super().get_serializer_class()
+
+    @action(methods=['get'], detail=True)
+    def retrieve_by_source_id(self, request, source_id):
+        pep = get_object_or_404(self.queryset, source_id=source_id)
```

Comment:
ohh i missed this

use self.get_queryset() instead of self.queryset
```suggestion
        pep = get_object_or_404(self.get_queryset(), source_id=source_id)
```

Reply:
done